packages/design-system/src/List/List.tsx

```diff
@@ -0,0 +1,187 @@
+      <TooltipTextWrapper ref={ref}>
+        <Text {...props} className={clsx("text-overflow", props.className)}>
+          {props.children}
+        </Text>
+      </TooltipTextWrapper>
+    </Tooltip>
+  );
+}
+
+function ListItem(props: ListItemProps) {
+  const {
+    description,
+    descriptionType = "block",
+    endIcon,
+    hasError,
+    size = "md",
+    startIcon,
+    title,
+  } = props;
+  const isBlockDescription = descriptionType === "block";
+
+  const listItemhandleKeyDown = (e: React.KeyboardEvent) => {
+    switch (e.key) {
+      case "Enter":
+      case " ":
+        props.onClick();
+        break;
+    }
+  };
+
+  const endIconhandleKeyDown = (e: React.KeyboardEvent) => {
+    e.stopPropagation();
+
+    if (props.onEndIconClick) {
+      switch (e.key) {
+        case "Enter":
+        case " ":
+          props.onEndIconClick();
+          break;
+      }
+    }
+  };
+
+  return (
+    <Wrapper>
+      <StyledListItem
+        data-disabled={props.isDisabled || false}
+        data-selected={props.selected}
+        endIcon={props.endIcon}
+        isBlockDescription={isBlockDescription}
+        onClick={props.onClick}
+        onKeyDown={listItemhandleKeyDown}
+        size={size}
+        tabIndex={props.isDisabled ? -1 : 0}
+      >
+        <ContentTextWrapper className="content-text-wrapper">
+          <StartIconWrapper>
+            {startIcon && (
+              <Icon
+                color={hasError ? "var(--ads-v2-color-fg-error)" : undefined}
+                name={startIcon}
+                size={size}
+              />
+            )}
+          </StartIconWrapper>
+          <InlineDescriptionWrapper>
+            <DescriptionWrapper>
+              <TextWithTooltip
+                className={ListItemTitleClassName}
+                color={hasError ? "var(--ads-v2-color-fg-error)" : undefined}
+              >
+                {title}
+              </TextWithTooltip>
+              {isBlockDescription && description && (
+                <TextWithTooltip
+                  className={ListItemBDescClassName}
+                  color="var(--ads-v2-color-fg-muted)"
+                  isMultiline
+                  kind="body-s"
+                >
+                  {description}
+                </TextWithTooltip>
+              )}
+            </DescriptionWrapper>
+            {!isBlockDescription && description && (
+              <TextWithTooltip
+                className={ListItemIDescClassName}
+                color="var(--ads-v2-color-fg-muted)"
+                kind="body-s"
+              >
+                {description}
+              </TextWithTooltip>
+            )}
+          </InlineDescriptionWrapper>
+        </ContentTextWrapper>
+      </StyledListItem>
+      {endIcon && (
+        <EndIconWrapper>
+          <Button
+            isDisabled={props.isDisabled}
+            isIconButton
+            kind="tertiary"
+            onClick={(e) => {
```

Comment:
Please add a handler function and avoid using anonymous function because it is not persistent when the component re-renders. This helps improving performance.